File: common/src/main/java/mconi/common/sim/subsystem/AtmosphereSubsystem.java
package mconi.common.sim.subsystem;

import mconi.common.sim.OniAtmosphereKernel;

public class AtmosphereSubsystem implements SimulationSubsystem
{
	private final OniAtmosphereKernel kernel = new OniAtmosphereKernel();

	@Override
	public String id()
	{
		return "atmosphere";
	}

	@Override
	public void run(SimulationContext context)
	{
		kernel.run(context.grid(), context.config());
	}
}

File: common/src/main/kotlin/mconi/common/sim/subsystem/AtmosphereSubsystem.kt
package mconi.common.sim.subsystem

import mconi.common.sim.OniAtmosphereKernel

class AtmosphereSubsystem : SimulationSubsystem {
    private val kernel = OniAtmosphereKernel()

    override fun id(): String = "atmosphere"

    override fun run(context: SimulationContext) {
        kernel.run(context.level(), context.config())
    }
}
